From 482d114d38352b0842557976bbd0ba78ec394c77 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 2 Apr 2015 11:39:20 +0200
Subject: [PATCH 01/49] base: make MSVC happy with __PRETTY_FUNCTION__

In patches further down the line, we emit some debug statements to
stdout/stderr for developers, and do so with the function name in it.
To cope with compilation under the Microsoft compiler, add an alias
from __PRETTY_FUNCTION__ to the respective constructs cl.exe
understands.
---
 base/logging.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/base/logging.h b/base/logging.h
index cc0a5aa..9cf8102 100644
--- a/base/logging.h
+++ b/base/logging.h
@@ -912,6 +912,11 @@ inline std::ostream& operator<<(std::ostream& out, const std::wstring& wstr) {
 #define NOTIMPLEMENTED_MSG "NOT IMPLEMENTED"
 #endif
 
+#ifdef _MSC_VER
+#	define __PRETTY_FUNCTION__ __FUNCSIG__
+#	define __func__ __FUNCSIG__
+#endif
+
 #if NOTIMPLEMENTED_POLICY == 0
 #define NOTIMPLEMENTED() EAT_STREAM_PARAMETERS
 #elif NOTIMPLEMENTED_POLICY == 1
-- 
2.4.3

